### PR TITLE
close #2636: added json schemas for Aspire.Hosting.AppHost and Aspire.Hosting.Azure

### LINKED ISSUE #2636: [WebToolsE2E][Aspire] There is no intellisense of 'Parameters' in the appsettings.json file.
**REGRESSION INFO**: New feature on Aspire 8.0 Preview 4

**INSTALL STEPS**

1. Clean machine: Win11 x64 23h2 ENU
2. Install 17.10.0 Preview 2.0 [34701.141.d17.10] (Includes Aspire 8.0.0-preview.4.24129.7)
3. Apply NuGet Feeds

**REPRO STEPS**

1. New Project > ASP.NET Core Empty project > 8.0 > check 'Enlist in Aspire Orchestration'
2. Right-click empty project > Add > Aspire component > install Aspire.Azure.Security.KeyVault
5. In the appsettings.json file of the empty project, type "" in the top of the file.

**Notes**:

1. It will meet [Bug 1985774](https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_workitems/edit/1985774) when adding the aspire component.

**ACTUAL**
No intellisense of 'Parameters' shows
image

**EXPECTED**
Should contain 'Parameters' in the intellisense

### PR DESCRIPTION
**Pull request by vladimir-shirmanov**
Description
fixed an issues with intellisense inside appsettings.json when using Aspire.Hosting.AppHost or Aspire.Hosting.Azure

Changes:
* added AspireAppHostConfiguration.json file and included it into Aspire.Host.AppHosting nuget package
* added src/Aspire.Hosting.Azure/AspireAzureConfigurationSchema.json and included it into Aspire.Host.Azure package

This will close #2636 
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/4912)

### CODE REVIEW

File: src/Aspire.Hosting.AppHost/Aspire.Hosting.AppHost.csproj [hunk 1]
```diff
@@ -16,7 +16,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <None Include="**/*.props;**/*.targets" Pack="true" PackagePath="%(RecursiveDir)%(Filename)%(Extension)" />
+    <None Include="**/*.props;**/*.targets;*.json" Pack="true" PackagePath="%(RecursiveDir)%(Filename)%(Extension)" />
```
Comment (eerhardt) [inline src/Aspire.Hosting.AppHost/Aspire.Hosting.AppHost.csproj:19]:
```suggestion
    <None Include="**/*.props;**/*.targets" Pack="true" PackagePath="%(RecursiveDir)%(Filename)%(Extension)" />
    <None Include="AspireAppHostConfiguration.json" Pack="true" />
```

Maybe we should just pack the one file we want, that way we don't accidently start including other .json files we don't intend to put in the nupkg.

File: src/Aspire.Hosting.Azure/build/Aspire.Hosting.Azure.targets [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<Project>
+  <ItemGroup>
+    <JsonSchemaSegment Include="$(MSBuildThisFileDirectory)..\AspireAzureConfigurationSchema.json"
+                       FilePathPattern="appsettings\..*json" />
```
Comment (eerhardt) [inline src/Aspire.Hosting.Azure/build/Aspire.Hosting.Azure.targets:4]: A lot of times these properties go into "User Secrets", which is a `secrets.json` file. This is a Regex, so I think we should also add `secrets.json` to it. @alexgav - did we ever hook up the FilePathPattern correctly in VS? 

```suggestion
                       FilePathPattern="appsettings\..*json|secrets.json" />
```

File: src/Aspire.Hosting.Azure/build/Aspire.Hosting.Azure.targets [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<Project>
```
Comment (eerhardt) [inline src/Aspire.Hosting.Azure/build/Aspire.Hosting.Azure.targets:1]: I think we only need a .targets file in `buildTransitive`. We shouldn't need the other 2 targets files. Just move this logic in there.

File: src/Aspire.Hosting.Azure/Aspire.Hosting.Azure.csproj [hunk 4]
```diff
@@ -19,11 +19,15 @@
     <Compile Include="..\Aspire.Hosting\Dcp\Process\ProcessUtil.cs" Link="Provisioning\Utils\ProcessUtil.cs" />
     <Compile Include="..\Shared\Cosmos\CosmosConstants.cs" Link="Shared\CosmosConstants.cs" />
   </ItemGroup>
-
+
```
Comment (eerhardt) [inline src/Aspire.Hosting.Azure/Aspire.Hosting.Azure.csproj:22]: (nit) this can be reverted.

**Review by eerhardt — COMMENTED** [review]
Thanks for the contribution, @vladimir-shirmanov!

File: src/Aspire.Hosting.AppHost/Aspire.Hosting.AppHost.csproj (hunk 1 above)
Comment (vladimir-shirmanov) [reply src/Aspire.Hosting.AppHost/Aspire.Hosting.AppHost.csproj]: sure, fixed it

File: src/Aspire.Hosting.Azure/build/Aspire.Hosting.Azure.targets (hunk 2 above)
Comment (vladimir-shirmanov) [reply src/Aspire.Hosting.Azure/build/Aspire.Hosting.Azure.targets]: I tried to add it, but it didn't work for me. So I skipped for now

File: src/Aspire.Hosting.Azure/build/Aspire.Hosting.Azure.targets (hunk 3 above)
Comment (vladimir-shirmanov) [reply src/Aspire.Hosting.Azure/build/Aspire.Hosting.Azure.targets]: sure, removed all unnecessary ones

File: src/Aspire.Hosting.Azure/Aspire.Hosting.Azure.csproj (hunk 4 above)
Comment (vladimir-shirmanov) [reply src/Aspire.Hosting.Azure/Aspire.Hosting.Azure.csproj]: ah sorry, removed it as well

**Comment by vladimir-shirmanov** [comment]
@eerhardt  thank you for the feedback. Did the changes you asked for

**Review by eerhardt — APPROVED** [review]
Nice work @vladimir-shirmanov! Thank you for this contribution. This looks good to me. I'm setting it to 'auto-merge' when CI passes.

**Pull request merged**